test/evp_extra_test2.c

```diff
@@ -346,10 +346,12 @@ static int test_dh_tofrom_data_select(void)
           && TEST_int_gt(EVP_PKEY_generate(gctx, &key), 0)
           && TEST_true(do_pkey_tofrom_data_select(key, "DHX"));
 # ifndef OPENSSL_NO_DEPRECATED_3_0
-    dhkey = EVP_PKEY_get0_DH(key);
-    ret = ret && TEST_ptr(dhkey);
-    ret = ret && TEST_ptr(privkey = DH_get0_priv_key(dhkey))
-              && TEST_int_le(BN_num_bits(privkey), 225);
+    if (key != NULL) {
+        dhkey = EVP_PKEY_get0_DH(key);
+        ret = ret && TEST_ptr(dhkey)
```

Comment:
IMO putting the dhkey assignment into this TEST_ptr() call would work too and be more concise. But OK as is.

Reply:
There will be an error message if `key == NULL`, so a TEST_ptr call is kind of irrelevant.

Reply:
There is the `ret = ret &&` so you could drop the `if (key != NULL)` around these calls.

Reply:
Two error messages could confuse.

Reply:
There would not be as the ret would be false already.

Reply:
`ret = ret && TEST_ptr(dhkey = EVP_PKEY_get0_DH(key));`

Reply:
Okay, this makes sense.

I'm in a world of my own today :(